pyopia/background.py

```diff
@@ -284,7 +327,10 @@ def __call__(self, data):
             data['skip_next_steps'] = True
             return data
 
-        data['im_corrected'] = correct_im_accurate(data['imbg'], data[self.image_source])
+        if self.divide_bg:
+            data['im_corrected'] = correct_im_accurate_divide(data['imbg'], data[self.image_source])
+        else:
+            data['im_corrected'] = correct_im_accurate(data['imbg'], data[self.image_source])
```

Comment:
Perhaps it would be simpler to pass a flag to the existing "correct_im_accurate" to indicate which background correction method to use? Then have the if inside that function. This avoids the need for a new function with some duplicate code.

Reply:
Nice, I will revise the code